src/transformStateWithClones.js

```diff
@@ -8,6 +8,51 @@
  */
 function transformStateWithClones(state, actions) {
   // write code here
+  const stateClone = { ...state };
+  const object = [];
+
+  for (let i = 0; i < actions.length; i++) {
+    switch (actions[i].type) {
+      case 'addProperties':
+        object.push(addProperties(stateClone, actions[i].extraData));
+
+        break;
+      case 'removeProperties':
+        object.push(removeProperties(stateClone, actions[i].keysToRemove));
+        // actions[i].keysToRemove.forEach(key => delete stateClone[key]);
+        break;
+      case 'clear':
+        // Object.keys(state).forEach(key => delete state[key]);
+        object.push(clearProperties(stateClone));
```

Comment:
it's confusing because you pushing to array result of `clearProperties` , so better rename or remove this function